addons/l10n_ro_edi_stock/models/stock_picking.py

```diff
@@ -431,26 +431,38 @@ def _l10n_ro_edi_stock_validate_data(self, data: dict):
         if not data['company_id'].l10n_ro_edi_access_token:
             errors.append(_('Romanian access token not found. Please generate or fill it in the settings.'))
 
+        # carrier for internal pickings
+        carrier_valid = True  # for non-internal pickings the carrier has already been checked during validation of the picking
+        if data['picking_type_id'].code == 'internal':
+            # Carrier is required when sending an eTransport document
+            if not data['carrier_id']:
+                errors.append(_("The delivery carrier is missing."))
+                carrier_valid = False
+            elif not data['carrier_id'].l10n_ro_edi_stock_partner_id:
+                errors.append(_("The delivery carrier is missing the partner field value."))
+                carrier_valid = False
+
```

Comment:
```suggestion
        # Carrier and carrier partner are required when sending an eTransport document
        partner = self.env['res.partner']
        if not data['carrier_id']:
            errors.append(_("The delivery carrier is missing."))
        elif not data['carrier_id'].l10n_ro_edi_stock_partner_id:
            errors.append(_("The delivery carrier is missing the partner field value."))
        else:
            partner = data['carrier_id'].l10n_ro_edi_stock_partner_id
```

Feel like this more readable and easier to manage in the future, what you think ? 

Furthermore, if we know that for non-internal transfer, the courier partner is set, I don't think it's a huge gain to create a if statement just for it. Quid if someone misused your API ? I think it's better to be safer and not "consider" non-internal transfers checked by default, for such little gain.